### ISSUE TITLE
facebook check

### ISSUE THREAD

**Issue by modinson** [keybase/client#14744]
keybase.io can't check my facebook post. First of all, when I posted the link on Facebook I got a 500 server error despite of this error the post is on my facebook account.

https://www.facebook.com/christoph.merta/posts/10213722788369406

**Comment by Ultrabenosaurus** [comment]
I've been having the same issue all day. Same with Reddit, which I'll make a new issue for if one doesn't already exist.

**Comment by immae** [comment]
Yes same for me, both with reddit and facebook.

For reddit, the issue is that the text doesn’t get into the template, only the title. If you take the text from the url and put it as text in the box (correctly url-decoded), then it works alright.

For facebook the error 500 is with the redirect url

**Comment by matsest** [comment]
Same here with Facebook. 500 redirection error after posting. Something with the FB post/template that has been changed?

**Comment by maxtaco** [comment]
Something is borked, we're looking into it, thanks!

**Comment by maxtaco** [comment]
FB killed a key permission on our app --- to get the post ID of your public post on Facebook. We've asked for a rereview to get the permission back.

**Comment by mhyzon** [comment]
Ah, cool.  Was just coming here tosay the same thing.  Glad you're on it!

**Comment by themann00** [comment]
Any update here?
I revoked mine, because it said it was pending- and I didn't know it was a bug. Now my FB profile link is gone from my Keybase.io profile. I'd like to get it back- even if it is pending for now.

**Comment by danskky** [comment]
It sounds more like a good news! I guess you're not selling enough user info them. Anyway, all useful services remained! :)

**Comment by ErikHumphrey** [comment]
Maybe it's time to read users' public PGP Public Key field instead? I suggested it before (not on GitHub) but can't remember why it was considered a bad idea.

**Comment by maxtaco** [comment]
There's no convenient way to step the user through this process and therefore no one will use it.

**Comment by jcolson** [comment]
I second using facebook's already public PGP field.

**Comment by TW1920** [comment]
@maxtaco Is there any plan to add it again at a later point? Or is there no chance to solve this?

**Comment by Daniel15** [comment]
@maxtaco Do you need any help from the Facebook side? Were there any details in the rejection email?

**Comment by BHydden** [comment]
@maxtaco surely we could just publically post a message that was signed with our private key to easily verify with our public key that we wrote it and pair the accounts?

**Comment by maxtaco** [comment]
@BHydden it's not that easy, since FB gives us no capacity to scan public feeds to figure out where the post is.

**Comment by BHydden** [comment]
Oh, I see what the issue is now... hmm, and the end-user can't supply you with the link manually during the linking process? Does it have to be scanned for?

**Comment by maxtaco** [comment]
Right. We'd need to instruct users on how to open the document inspector and read the post ID out of the feed. It's a shitshow. Our little app was only needed to do that automatically on behalf of the user (only scanning *public* posts, mind you), but Facebook took it down.

**Comment by BHydden** [comment]
Yeah, I see the issue now. On one hand you've got tech-competent people saying why is this even an issue, but on the other hand once you open that door you'll be flooded with "I don't understand how to get this link" complaints... shame FB are being dicks about the whole thing...

**Comment by ErikHumphrey** [comment]
@Daniel15 Consider adding an API endpoint to read the "PGP Public Key" found on a Facebook user profile at About -> Contact and Basic Info -> Contact Information -> PGP Public Key.

Also consider adding a PGP Public Key to Pages if it's not a thing yet along with an endpoint.

**Comment by Daniel15** [comment]
> We'd need to instruct users on how to open the document inspector and read the post ID out of the feed

It's usually in the permalink URL. Could you tell users to go to the permalink for the post (click on the timestamp) then copy and paste the URL? Or right-click the timestamp and "Copy Link"

>  Consider adding an API endpoint to read the "PGP Public Key" found on a Facebook user profile at About -> Contact and Basic Info -> Contact Information -> PGP Public Key.

It looks like it's already there as a `public_key` field as per https://developers.facebook.com/docs/graph-api/reference/user/. However if I try to fetch it with a test app, it doesn't appear in the response. Will see if I can follow up - perhaps some special permission is needed.

**Comment by BHydden** [comment]
> (click on the timestamp)

TIL ^ 😮 nice one! 👍 



> Will see if I can follow up

Awesome! Thanks 😄 Adding a public PGP key is easy, just copy-paste from keybase into the menu section mentioned using a desktop (can't be done from mobile app, but then neither can adding ... so it seems the last hurdle is keybase being allowed to crawl facebook for listed public keys (I believe) so if @Daniel15  can get that API working then @maxtaco  should be able to re-implement Facebook pairing 😃

**Comment by Daniel15** [comment]
> TIL ^ 😮 nice one! 👍

It's a pretty standard way to get a permalink. Lots of sites do the same thing - Even here on GitHub, the comment timestamps are all permalinks 😃

**Comment by BHydden** [comment]
> Even here on GitHub, the comment timestamps are all permalinks

And here I've been using the `...` menu like a chump! :open_mouth:
Duly noted! I shall henceforth click on every time stamp I see to check if it is a sneaky hidden perma-link or not! :smile: love finding new standards I never knew about that turn out to be super useful / practical! :heart:

**Comment by ErikHumphrey** [comment]
@BHydden It would be nice to prevent having to post a public status update if possible; most Keybase proofs look spammy/unsightly.

**Comment by BHydden** [comment]
@ErikHumphrey I dunno, I kinda like it for the advertising potential. I first stumbled upon keybase by seeing a bunch of identity gists on github and wondering what that was all about.

**Comment by kimcosmos** [comment]
Or create a keybase facebook account. Then user presses a copy button and pastes a pgp message to keybase on facebook. Dont even need to friend keybase i think. Facebook api scripting should be able to read it. api polling has a 5 minute maximum frequency from memory. Just tell them to be patient

**Comment by nick4fake** [comment]
Are there any news?

**Comment by BHydden** [comment]
TMK Keybase hasn't been actively developed since they were bought by Zoom
